mp/src/game/client/sdk/vgui/sdk_spectatorgui.cpp

```diff
@@ -254,6 +254,59 @@ void CSDKSpectatorGUI::Paint()
 		surface()->DrawSetTextFont( vgui::scheme()->GetIScheme(vgui::scheme()->GetScheme( "ClientScheme" ))->GetFont( "Default" ) );	//reset the font, draw icon can change it
 		surface()->DrawUnicodeString( sButtons.c_str(), vgui::FONT_DRAW_NONADDITIVE );
 	}
+
+	int spec = pLocalPlayer->GetObserverMode();
+	if (ShouldShowPlayerLabel(spec) || spec == OBS_MODE_ROAMING)
+	{
+		// The color to be used
+		surface()->DrawSetTextColor(Color(255, 255, 255, 128));
+
+		// The font to be used
+		vgui::HFont font = vgui::scheme()->GetIScheme(vgui::scheme()->GetScheme("ClientScheme"))->GetFont("SpectatorNames");
+		surface()->DrawSetTextFont(font);
+
+		for (int i = 1; i <= gpGlobals->maxClients; i++)
+		{
+			CSDKPlayer* pPlayer = ToSDKPlayer(UTIL_PlayerByIndex(i));
+			if (!pPlayer)
+				continue;
+
+			if (pPlayer == pLocalPlayer)
+				continue;
+
+			if (spec != OBS_MODE_ROAMING && pPlayer == pObserved)
+				continue;
+
+			if (!pPlayer->IsVisible())
+				continue;
+
```

Comment:
good point, gotta see check if you can see spectators with this :D
Otherwise, it's probably covered by the visibility check